### ISSUE TITLE
Request: Folder whitelist + blacklist

### ISSUE THREAD

**Issue by Ketrel** [kollerlukas/Camera-Roll-Android-App#142]
Could an option be added in which only specific folders are included?

(The blacklisted folders could still apply with higher priority)

**Comment by kollerlukas** [comment]
Blacklisting is already possible. You can go into Settings and exclude specific Directories.

**Comment by Ketrel** [comment]
I know, I'm talking about adding whitelisting.

So in theory, I could say

ONLY include 
    /sdcard/Pictures
    /sdcard/Snapchat

And exclude  
    /sdcard/Pictures/<App that saves here without the ability to remove>/

(That was the only reason I mentioned blacklisting, in addition to whitelisting)

**Comment by nicolas-raoul** [comment]
I would love whitelisting.

I use the app to show only one folder, so whitelisting would be a one-time operation.
Blacklisting forces me to fix settings every time I install an application that stores media in its own folder (which happens very often, so many apps do this, for instance WhatsApp or Foodie)

@Ketrel How about renaming this issue to "Request: Folder whitelist" to avoid confusion?

Thanks! :-)

**Comment by Ketrel** [comment]
@nicolas-raoul renaming it that would be inaccurate because a big part of what I'm asking is that the blacklist exist as well and supersede the whitelist.

That would allow cases like

whitelist /sdcard/pictures
but blacklist /sdcard/pictures/that-one-app

**Comment by IzzySoft** [comment]
@Ketrel that doesn't contradict. @nicolas-raoul just suggested renaming the *issue* – and "Request: folder whitelist" would match both of your intentions. Nicolas didn't want to replace the blacklist by a whitelist. What you describe is fully in line with what he wants. Reference: [This is what Nicolas wants to achieve](https://softwarerecs.stackexchange.com/q/50858/185). So he'd use just the first step of your request *without setting up the second step.* But in general, both of you want a whitelist feature :wink:

But I agree: renaming the issue isn't really needed :smile_cat:

**Comment by IzzySoft** [comment]
Thought so, @Ketrel – but IMHO the "color" is not necessarily the criterion for precedence. The "more specific" one could be taken as such as well. You could e.g. wish to *blacklist* `/sdcard/pictures` (so all its sub-directories would be blacklisted) but *whitelist* `/sdcard/pictures/important_app`. In this case, the blacklist would match the latter directory, but the whitelist entry being "more specific" and taking precedence. The case you described would work as well by this "algorithm".

@kollerlukas what do you think?

**Comment by Ketrel** [comment]
@IzzySoft I agree.  The most specific rule having the highest priority is the best logic for handling that IMO.
